### DISCUSSION TITLE
Code server taking a long time to load in runs

### DISCUSSION THREAD

**Discussion by brunobento-delfos** [dagster-io/dagster#19481] [Q&A]
Every time I run a job or an asset, it seems like it loads all the code server again before executing it.
This is really bad because we have a repository that takes ~30s to load.

I simulated this by creating a simple code with a 10 second sleep on the Definitions file.
Every time, between ENGINE_EVENT and RUN_START events, it takes exactly 10s to start because it is loading the codeserver.
If you keep re-executing the job forever you'll notice this behavior.
![image](https://github.com/dagster-io/dagster/assets/153758476/cf01cec7-5389-4e4e-81ee-169185d6e690)

It looks like each run is in an isolated environment. Is there any way to disable this behavior?

I demonstrated a simple case using docker-compose in this [link](https://github.com/brunobento-delfos/dagster-code-server-sleep).

**Comment by dpeng817** [comment]
The default run launcher will launch each run in a separate process, thus requiring the reload of the module. You would have to write a custom thread-based run launcher in order to achieve the behavior you're looking for here.

**Reply by brunobento-delfos** [reply]
Thanks for the answer @dpeng817 !
